spark/src/main/scala/io/delta/sharing/spark/RemoteDeltaLog.scala

```diff
@@ -173,13 +180,21 @@ private[sharing] object RemoteDeltaLog {
 class RemoteSnapshot(
     tablePath: Path,
     client: DeltaSharingClient,
-    table: DeltaSharingTable) extends Logging {
+    table: DeltaSharingTable,
+    versionOf: Option[Long] = None) extends Logging {
 
   protected def spark = SparkSession.active
 
   lazy val (metadata, protocol, version) = {
-    val tableMetadata = client.getMetadata(table)
-    (tableMetadata.metadata, tableMetadata.protocol, tableMetadata.version)
+    if (versionOf.isEmpty) {
+      val tableMetadata = client.getMetadata(table)
+      (tableMetadata.metadata, tableMetadata.protocol, tableMetadata.version)
+    } else {
+      // getMetadata doesn't support the parameter: versionOf
+      // Leveraging getFiles to get the metadata, so setting the limitHint to 1 for efficiency.
+      val tableFiles = client.getFiles(table, Nil, Some(1L), versionOf)
```

Comment:
Should we add a method to the client: getMetadata(versionOf) and use it for getting protocol/metadata?

Reply:
Perhaps a util method then?